src/web/common.rkt

```diff
@@ -77,118 +78,166 @@
+    ["MATLAB"
+     (define comma-joined (comma-join vs))
      (format "~a = num2cell(sort([~a])){:}" comma-joined comma-joined)]
-    [(list 'abs x)
-     (format "~a = abs(~a)" x x)]))
-
-(define (preprocess->tex preprocess)
-  (match preprocess
-    [(list 'sort variables ...)
-     (define comma-joined (string-join (map ~a variables) ", "))
+    ["TeX"
+     (define comma-joined (comma-join vs))
      (format "[~a] = \\mathsf{sort}([~a])\\\\" comma-joined comma-joined)]
-    [(list 'abs x)
-     (format "~a = |~a|\\\\" x x)]))
-
-(define (preprocess->default preprocess)
-  (match preprocess
-    [(list 'sort a b)
-     (format sort-note (format "~a and ~a" a b))]
-    [(list 'sort variables ...)
-     (format
-      sort-note
-      (string-join (map ~a variables) ", " #:before-last ", and "))]
-    [(list 'abs x)
-     (format abs-note x)]))
-
-(define sort-note
-  "NOTE: ~a should be sorted in increasing order before calling this function.")
+    [_
+     (match vs
+       [(list x y) (format sort-note (format "~a and ~a" x y))]
+       [(list vs ...)
+        (format
+         sort-note
+         (string-join (map ~a vs) ", "
+                      ;; "Lil Jon, he always tells the truth"
+                      #:before-last ", and "))])]))
 
-(define abs-note "NOTE: ~a should be positive before calling this function")
-
-(define languages
-  `(("FPCore" "fpcore" ,(λ (c i) (fpcore->string c)) ,preprocess->default)
-    ("C" "c" ,core->c ,preprocess->c)
-    ("Fortran" "f03" ,core->fortran ,preprocess->default)
-    ("Java" "java" ,core->java ,preprocess->java)
-    ("Python" "py" ,core->python ,preprocess->python)
-    ("Julia" "jl" ,core->julia ,preprocess->julia)
-    ("MATLAB" "mat" ,core->matlab ,preprocess->matlab)
-    ("Wolfram" "wl" ,core->wls ,preprocess->default)
-    ("TeX" "tex" ,(λ (c i) (core->tex c)) ,preprocess->tex)))
+(define (format-less-than-condition variables)
+  (string-join
+   (for/list ([a (in-list variables)]
+              [b (in-list (cdr variables))])
+     (format "~a < ~a" a b))
+   " && "))
 
-(define (program->tex prog ctx #:loc [loc #f])
-  (define prog* (program->fpcore prog ctx))
-  (if (supported-by-lang? prog* "tex")
-      (core->tex prog* #:loc (and loc (cons 2 loc)) #:color "blue")
-      "ERROR"))
+(define (comma-join vs)
+  (string-join (map ~a vs) ", "))
 
-(define (add-preprocessing-tex preprocess-lines output)
-  (format "\\begin{array}{l}\n~a\\\\\n~a\\end{array}\n"
-          preprocess-lines output))
+(define sort-note
+  "NOTE: ~a should be sorted in increasing order before calling this function.")
 
-(define (render-program preprocesses expr ctx #:ident [identifier #f] #:pre [precondition '(TRUE)])
+(define (render-program expr ctx #:ident [identifier #f] #:pre [precondition '(TRUE)] #:instructions [instructions empty])
   (define output-repr (context-repr ctx))
+  (match-define (cons expr* ctx*)
+    (foldl
+     (match-lambda* [(list i (cons e c)) (combine-fpcore-instruction i e c)])
```

Comment:
Lol bro. It's fine, I'll merge it, but c'mon, never use folds. This all is just an append + a map anyway.